lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -477,23 +477,53 @@ AMDMfmaEncodingAttr::toLinearLayout(ArrayRef<int64_t> shape) const {
   return combineCtaCgaWithShape(tileLayout, getCTALayout(), shape);
 }
 
+LinearLayout chooseLLDsReadB64TrLayout(Attribute enc, ArrayRef<int64_t> shape,
+                                       int32_t elemBitWidth) {
+  using BaseTy = std::vector<std::vector<int32_t>>;
+
```

Comment:
This is a cool trick; it took me a while to fully understand it. :) Having some high level comments would make it easier to following. Like mentioning that this is deducing the ds_read_b64_tr instruction layout from an input dot operand like layout. There is a correlation between the dot operand like layout and the layout ds_read_b64_tr needed to read the shared memory. Then we can define a "base unit" as a 16(M/N)x128-bit(K) element. We "remap" element ownership within every such base unit--originally for the input dot operand like layout each lane owns 128-bit along K and then different lanes own different M/N; after remapping each lane owns 64-bit along M/N first and then different lanes are tiled along K..

Reply:
If with some graph or example it'd be even better!